oak_functions/loader/src/server.rs

```diff
@@ -43,6 +47,90 @@ type AbiPointerOffset = u32;
+/// the policy is failed.
+const MIN_RESPONSE_SIZE: u32 = 90;
+
+/// A policy describing limits on the size of the response and response processing time to avoid
+/// side-channel leaks.
+#[derive(Deserialize, Debug, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct Policy {
+    /// A fixed size for responses returned by the trusted runtime.
+    ///
+    /// If the response computed by the Wasm module is smaller than this amount, it is padded with
+    /// additional data before encryption in order to make the payload size exactly this size. If
+    /// the response is larger than this amount, the trusted runtime discards the response and
+    /// instead sends a response of exactly this size to the client, containing an error message
+    /// indicating the failure.
+    pub constant_response_size_bytes: u32,
+    /// A fixed response time.
+    ///
+    /// Similar to the previous one, but controls the amount of time the function is allowed to run
+    /// for. If the function finishes before this time, the response is not sent back until the
+    /// time is elapsed. If the function does not finish within this deadline, the trusted runtime
+    /// sends a response to the client containing an error message indicating the failure. The size
+    /// of this response is equal to the size specified by the previous parameter.
+    #[serde(with = "humantime_serde")]
+    pub constant_processing_time: Duration,
+}
+
+impl Policy {
+    pub fn validate(&self) -> anyhow::Result<()> {
+        match self.constant_response_size_bytes >= MIN_RESPONSE_SIZE {
+            true => Ok(()),
+            false => anyhow::bail!("Response size is too small"),
+        }
+    }
+
+    fn get_error_response(&self, reason: &str) -> anyhow::Result<Response<Body>> {
+        // We include the `oak-policy-padding` in all responses. So first estimate the size of
+        // the response including this header name. The diff between this size and
+        // `self.constant_response_size_bytes` gives the required padding, that will be added as the
+        // value of the header.
+        let body = format!("Reason: {}\n", reason);
+        let estimates_size = estimate_error_response_size(body.clone())?;
+        let required_padding = self.constant_response_size_bytes as i64 - estimates_size;
+
+        if required_padding < 0 {
+            anyhow::bail!("constant_response_size_bytes specified by the policy is too small.")
+        } else {
+            let padding_bytes: Vec<u8> = repeat(b'0').take(required_padding as usize).collect();
+            http::response::Builder::new()
+                .status(StatusCode::INTERNAL_SERVER_ERROR)
+                .header(POLICY_PADDING_HEADER, std::str::from_utf8(&padding_bytes)?)
+                .body(body.into())
+                .context("couldn't create response")
+        }
+    }
+
+    /// Pads the response by adding a number of zeros, to make the total size of the response equal
```

Comment:
I would either leave the description of exactly what the padding is unspecified ("adding a number of unspecified characters"), or actually specify the character (though I think this is unnecessarily constraining the contract of this function, the specific character seems an impl detail and we should be able to change it without having to update this doc string.